packages/manager/modules/hub/src/components/product-list/controller.js

```diff
@@ -0,0 +1,92 @@
+import find from 'lodash/find';
+import get from 'lodash/get';
+import keysIn from 'lodash/keysIn';
+import map from 'lodash/map';
+
+import { PRODUCTS } from './config';
+
+export default class ManagerHubBillingProductList {
+  /* @ngInject */
+  constructor($http, $translate) {
+    this.$http = $http;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    if (!this.product) {
+      throw new Error('Missing required product attribute');
+    }
+    this.isLoading = true;
+
+    this.route = get(PRODUCTS, `${this.product}.route`, this.product);
+    this.columns = get(PRODUCTS, `${this.product}.columns`);
+    this.formater = get(PRODUCTS, `${this.product}.formater`);
+
+    this.columns = map(this.columns, (column) => ({
+      ...column,
+      title: this.$translate.instant(column.title),
+      filterable: 'filterable' in column ? column.filterable : true,
+      sortable: 'sortable' in column ? column.sortable : true,
+    }));
+
+    this.endpoint = this.route.replace(/\/\{[^}]+\}/, '');
+    [, this.resource] = this.route.match(/\{([^}]+)\}/);
+
+    return this.fetchProducts()
+      .then((products) => this.fetchProductsBilling(products))
+      .then((products) => {
+        if (!this.columns.length) {
+          this.columns = map(keysIn(products[0]), (key) => ({
+            title: key,
+            property: key,
+            filterable: true,
+            sortable: true,
+          }));
+        }
+        return products;
+      })
+      .then((products) => {
+        this.products = this.formater ? map(products, this.formater) : products;
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
+
+  fetchProducts() {
+    return this.$http
+      .get(this.endpoint, {
+        headers: {
+          'X-Pagination-Mode': 'CachedObjectList-Pages',
+          'X-Pagination-Size': 50000,
+        },
+      })
+      .then(({ data }) => data);
+  }
+
+  fetchProductsBilling(products) {
```

Comment:
I don't think the component should make any calls to add logic to its data. 
Also we already have this call made in the root state app

Reply:
We need to link the billing part to the product in order to display product's infos & service expiration on the same row.
I see that the hub is returning only the first 4 results of billing services. I have two options : making a call to /billing/autorenew/services here or adding an option parameter to the hub 2api endpoint to fetch all the billing infos. Either way i'll have to do some logic to link the billing service to the product.
What do you think?

Reply:
And i'll also have to make a call to either /billing/autorenew/services or /hub here because the root hub call in the app won't fetch all the billingServices

Reply:
@frenautvh The url is agreggated to `/services` call in 2API, It has been taken in account in https://github.com/ovh/manager/pull/2400